docs/sdks/client-sdks/javascript/assignments.mdx

```diff
@@ -0,0 +1,411 @@
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = "version-a"; 
+
+const variant = client.getStringAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+
+);
+
+// Use the variant value to determine which component to render
+if (variant === "version-a") {
+    return <versionAComponent />
+} else if (variant === "version-b") {
+    return <versionBComponent />
+}
+
+```
+
+## Boolean Assignments
+
+Boolean flags support simple on/off toggles. They're useful for simple, binary feature switches like blue/green deployments or enabling/disabling a new feature.
+
+```js
+import * as EppoSdk from "@eppo/js-client-sdk";
+
+const client = EppoSdk.getInstance();
+
+const flagKey = "flag-key-123";
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = false; 
+
+const variant = client.getBooleanAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+);
+
+// Use the variant value to determine which component to render
+if (variant) {
+    return <FeatureEnabledComponent />
+} else {
+    return <FeatureDisabledComponent />
+}
+```
+
+## JSON Assignments
+
+JSON flags work best for advanced configuration use cases. The JSON flag can include structured information such as 
+
+- the text of a marketing copy for a promotional campaign 
+- the address of a different hero image. 
+
+Using this pattern, a team can make minor changes to the copy and design of a website without having to go through an entire code release process.
+
+For example, the following code shows how to use a JSON flag to configure a campaign.
+
+```js
+import * as EppoSdk from "@eppo/js-client-sdk";
+
+const client = EppoSdk.getInstance();
+
+const flagKey = "flag-key-123";
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = {
+    hero: false,
+    heroImage: "placeholder.png",
+    heroTitle: "Placeholder Hero Title",
+    heroDescription: "Placeholder Hero Description"
+}; 
+
+const campaignJson = client.getJSONAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+);
+
+if (campaignJson !== null) {  
+    campaign.hero = true;
+    campaign.heroImage = campaignJson.heroImage;
+    campaign.heroTitle = campaignJson.heroTitle || "";
+    campaign.heroDescription = campaignJson.heroDescription || "";
+}
+
+// JSX component sets itself based on the campaignJson
+<Hero image={campaignJson.heroImage} title={campaignJson.heroTitle} description={campaignJson.heroDescription} />
+```
+
+## Integer and Numeric Assignments
+
+Integer and numeric assignments work the same way but return either an integer or a floating point number. These assignments are useful where you want to use a numeric value to drive business logic such as pricing on an item or a number of items to display in a list.
+
+```js
+import * as EppoSdk from "@eppo/js-client-sdk";
+
+const client = EppoSdk.getInstance();
+const flagKey = "flag-key-123";
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = 0; 
```

Comment:
Wouldn't it make more sense to have a `defaultNumberOfItems` and a separate `defaultPrice`?